artifactory/commands/transferfiles/delayedartifactshandler.go

```diff
@@ -123,12 +182,63 @@ func consumeDelayedArtifactsFiles(pcWrapper *producerConsumerWrapper, filesToCon
 		if err != nil {
 			return errorutils.CheckError(err)
 		}
-
-		log.Debug("Done handling delayed artifacts file: '" + filePath + "'")
+		log.Debug("Done handling delayed artifacts file: '" + filePath + "'. Deleting it...")
 	}
 	return nil
 }
 
+// Reads a delay file from a given path, parses and populate a given DelayedArtifactsFile instance with the file information
+func readDelayFile(path string) (DelayedArtifactsFile, error) {
+	// Stores the errors read from the errors file.
+	var delayedArtifactsFile DelayedArtifactsFile
+
+	fContent, err := os.ReadFile(path)
+	if err != nil {
+		return delayedArtifactsFile, errorutils.CheckError(err)
+	}
+
+	err = json.Unmarshal(fContent, &delayedArtifactsFile)
+	if err != nil {
+		return delayedArtifactsFile, errorutils.CheckError(err)
+	}
+	return delayedArtifactsFile, nil
+}
+
+// Gets a list of all delay files from the CLI's cache for a specific repo
+func getDelayFiles(repoKeys []string) (filesPaths []string, err error) {
+
+	dirPath, err := coreutils.GetJfrogTransferDelaysDir()
+	if err != nil {
+		return []string{}, err
```

Comment:
return []string{}, err --> return

The same goes for the other retuern statements in this function.